hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MemberHandshake.java

```diff
@@ -67,24 +67,30 @@ public MemberHandshake(byte schemaVersion,
                            Map<ProtocolType, Collection<Address>> localAddresses,
                            Address targetAddress,
                            boolean reply,
-                           UUID uuid,
-                           int planeIndex,
-                           int planeCount) {
+                           UUID uuid) {
         this.schemaVersion = schemaVersion;
         this.localAddresses = new EnumMap<>(localAddresses);
         this.targetAddress = targetAddress;
         this.reply = reply;
         this.uuid = uuid;
-        this.planeIndex = planeIndex;
-        this.planeCount = planeCount;
     }
 
-    public int getPlaneIndex() {
-        return planeIndex;
+    public MemberHandshake addOption(String key, Object value) {
+        options.put(key, "" + value);
+        return this;
+    }
+
+    public int getIntOption(String key, int defaultValue) {
+        String value = options.get(key);
+        return value == null ? defaultValue : Integer.parseInt(value);
     }
 
     public int getPlaneCount() {
-        return planeCount;
+        return getIntOption("planeCount", 1);
```

Comment:
minor: add `static final String`s for `planeCount` & `planeIndex` keys

Reply:
Fixed